post_processing_stages/motion_detect_stage.cpp

```diff
@@ -0,0 +1,190 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2021, Raspberry Pi (Trading) Limited
+ *
+ * motion_detect_stage.cpp - motion detector
+ */
+
+// A simple motion detector. It needs to be give a low resolution image and it
+// compares pixels in the current low res image again the value in the corresponding
+// location in the previous one. If it exceeds a threshold it gets counted as
+// "different". If enough pixels are different, that indicates "motion".
+// A low res image of something like 128x96 is probably more than enough, and you
+// can always subsample with hskip and vksip.
+
+// Because this gets run in parallel by the post-processing framework, it means
+// the "previous frame" is not totally guaranteed to be the actual previous one,
+// though in practice it is, and it doesn't actually matter even if it wasn't.
+
```

Comment:
Not something to do now, but I wonder if we should have a mode that ensures we use the "current" and "previous" frame only.  Would that be useful?

Reply:
I agree it would be nice to guarantee that the "previous" frame really is the actual genuinely previous one. But we're at the mercy of the multiple threads, which thread tries to run first, who gets the mutex first... that kind of thing. So I'm not entirely sure what the best way is to do that. Perhaps run it before we split into several threads? Does that over-complicate the post-processing idea? Open to suggestions on this one!

Reply:
Need to think about how to do this a bit more.  Given the postprocessing thread dispatch code ensures that we return results in order, I think we may be able to do some trickery with mutexes or other sync primitives...